### ISSUE TITLE
Compilation fails with GCC 8.4.

### ISSUE THREAD

**Issue by maxaehle** [InsightSoftwareConsortium/ITK#3643]
### Description
Compilation fails.

### Steps to Reproduce
Clone ITK into `ITK/src`, cd into `ITK/bin` and 
```
cmake ../src/ -DCMAKE_CXX_FLAGS:STRING=-std=c++11 -DBUILD_DOCUMENTATION=OFF -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF -DModule_RTK=ON -DRTK_USE_CUDA=OFF -DCMAKE_BUILD_TYPE=Release -DRTK_BUILD_APPLICATIONS=OFF
make
```

### Expected behavior
Successful compilation.

### Actual behavior
Compilation aborts with error message like this one:
```
[ 28%] Building CXX object Modules/Core/Common/src/CMakeFiles/ITKCommon.dir/__/itkBuildInformation.cxx.o
In file included from /software/gcc/8.4.0/include/c++/8.4.0/memory:80,
                 from /home/aehle/ITK/src/Modules/Core/Common/include/itkExceptionObject.h:22,
                 from /home/aehle/ITK/src/Modules/Core/Common/include/itkMacro.h:1368,
                 from /home/aehle/ITK/src/Modules/Core/Common/include/itkBuildInformation.h:21,
                 from /home/aehle/ITK/bin/Modules/Core/Common/itkBuildInformation.cxx:19:
/software/gcc/8.4.0/include/c++/8.4.0/bits/unique_ptr.h: In instantiation of ‘void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = itk::SubjectImplementation]’:
/software/gcc/8.4.0/include/c++/8.4.0/bits/unique_ptr.h:277:17:   required from ‘std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = itk::SubjectImplementation; _Dp = std::default_delete<itk::SubjectImplementation>]’
/home/aehle/ITK/src/Modules/Core/Common/include/itkObject.h:277:75:   required from here
/software/gcc/8.4.0/include/c++/8.4.0/bits/unique_ptr.h:79:16: error: invalid application of ‘sizeof’ to incomplete type ‘itk::SubjectImplementation’
  static_assert(sizeof(_Tp)>0,
                ^~~~~~~~~~~
```

### Reproducibility
The above error happens consistently in the environment described below.

### Versions
Current master branch, commit 171fb2ba33a87041f99328a2f26612ff33aa9cc8.

### Environment
Linux, CMake 3.20.2, **GCC 8.4.0**.  
(You can reproduce a similar environment with GCC 8.5.0, which also exhibits this problem, by building a Singularity container with this definition file: [singularity.def.txt](https://github.com/InsightSoftwareConsortium/ITK/files/9617025/singularity.def.txt).)

### Additional Information
The compilation error can be fixed by removing the initializer  "`{ nullptr }`" of [`Object::m_SubjectImplementation`](https://github.com/InsightSoftwareConsortium/ITK/blob/171fb2ba33a87041f99328a2f26612ff33aa9cc8/Modules/Core/Common/include/itkObject.h#L277). So maybe this problem is related to #3620.
 With GCC 9.4 the problem does not come up.

**Comment by N-Dekker** [comment]
> The compilation error can be fixed by removing the initializer "{ nullptr }" of [Object::m_SubjectImplementation](https://github.com/InsightSoftwareConsortium/ITK/blob/171fb2ba33a87041f99328a2f26612ff33aa9cc8/Modules/Core/Common/include/itkObject.h#L277).

@maxaehle Thanks, that's a very helpful suggestion, Max! The "{ nullptr }" initializer isn't really necessary for `unique_ptr`s, I just thought it might have improved code readability! 🤔 In this particular case, would an empty `{}` initializer still be OK for GCC 8.4?

**Comment by N-Dekker** [comment]
> cmake ../src/ -DCMAKE_CXX_FLAGS:STRING=-std=c++11

ITK now requires at least C++14. Does the compile error also occur when you do `-DCMAKE_CXX_FLAGS:STRING=-std=c++11`?

**Comment by N-Dekker** [comment]
Thanks again Max. Just reproduced at https://godbolt.org/z/zM8PM5eTe even with GCC 9.1:
```

#include <memory>

class SubjectImplementation;

class Object
{
  Object();
  ~Object();
  std::unique_ptr<SubjectImplementation> m_SubjectImplementation{ nullptr };
};

```

Output:
```
In file included from /opt/compiler-explorer/gcc-9.1.0/include/c++/9.1.0/memory:80,
                 from <source>:5:
/opt/compiler-explorer/gcc-9.1.0/include/c++/9.1.0/bits/unique_ptr.h: In instantiation of 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = SubjectImplementation]':
/opt/compiler-explorer/gcc-9.1.0/include/c++/9.1.0/bits/unique_ptr.h:289:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = SubjectImplementation; _Dp = std::default_delete<SubjectImplementation>]'
<source>:13:75:   required from here
/opt/compiler-explorer/gcc-9.1.0/include/c++/9.1.0/bits/unique_ptr.h:79:16: error: invalid application of 'sizeof' to incomplete type 'SubjectImplementation'
   79 |  static_assert(sizeof(_Tp)>0,
      |                ^~~~~~~~~~~
```

I'll prepare a pull request, probably just having your suggested workaround.
